base/workflow/R/run.write.configs.R

```diff
@@ -26,8 +26,10 @@ run.write.configs <- function(settings, write = TRUE, ens.sample.method = "unifo
                               posterior.files = rep(NA, length(settings$pfts)), 
                               overwrite = TRUE) {
   
-  con <- PEcAn.DB::db.open(settings$database$bety)
-  on.exit(PEcAn.DB::db.close(con))
+  con <-
```

Comment:
So the problem here is that the later code that uses `con` is not checking to see if the connection was successful. I'd move this whole chunk into the Posteriors code below that uses the con (a few ifs in), since we don't need the connection open unless those conditions are met, and then to add an explicit `logger.severe` if we need the database connection but don't have access to the DB

Reply:
In addition to what Mike said, I would implement this as a proper `tryCatch` statement:

```
tryCatch({
  con <- PEcAn.DB::db.open(...)
  on.exit(PEcAn.DB::db.close(con)
}, error = function(e) {
  PEcAn.logger::logger.severe("Connection requested, but failed to open with the following error: ", conditionMessage(e)
})
```